cns/restserver/internalapi.go

```diff
@@ -389,3 +389,61 @@ func (service *HTTPRestService) CreateOrUpdateNetworkContainerInternal(req *cns.
 
 	return returnCode
 }
+
+// GetHomeAz gets home AZ from cache or from nmagent
+func (service *HTTPRestService) GetHomeAz(ctx context.Context) cns.GetHomeAzResponse {
+	homeAz := service.readHomeAzCache()
+	if homeAz != "" {
+		return cns.GetHomeAzResponse{HomeAzResponse: nmagent.HomeAzResponse{HomeAz: homeAz}}
+	}
+	suppApisCache := service.readSupportedApisCache()
+	if len(suppApisCache) == 0 || !isAPISupportedByNMAgent(suppApisCache, getHomeAzAPIName) { // to account for if nma getting updated underneath
+		// Get NMAgent supported apis
+		supportedAPIs, err := service.nma.SupportedAPIs(ctx)
+		if err != nil {
+			returnMessage := fmt.Sprintf("[Azure CNS] Error. getHomeAz failed to get NMAgent supported Apis %v", err)
```

Comment:
IMO the return msg may be a bit misleading. It should indicate that the "xx" API call to NMA failed. xx - correct API name exposed by NMA to get supported APIs